presto-raptor/src/main/java/com/facebook/presto/raptor/metadata/MySqlShardDao.java

```diff
@@ -39,4 +40,14 @@
     @SqlBatch("INSERT IGNORE INTO deleted_shards (shard_uuid, delete_time)\n" +
             "VALUES (:shardUuid, CURRENT_TIMESTAMP)")
     void insertDeletedShards(@Bind("shardUuid") Iterable<UUID> shardUuids);
+
+    @SqlUpdate("DELETE FROM transactions\n" +
+            "WHERE end_time < :maxEndTime AND (\n" +
```

Comment:
I tried the query on one shard of and its completion time is comparable for the 2 versions. So, I think we can use a simpler version for now and use an optimized one later if necessary.